### ISSUE TITLE
Generalized configuration options

### ISSUE THREAD

**Issue by fehguy** [swagger-api/swagger-codegen#616]
Need to pass a generalized configuration object to the generator so the default values can be overridden.  Otherwise, everyone needs to recompile for every change.

**Comment by fehguy** [comment]
Suggest the following:

1) allow a json config file `--config @config.json`
2) allow each language to expose a separate set of options that are usable from the command-line

We will still have sensible defaults that won't be `Petstore`.

**Comment by who** [comment]
:+1:  I like this, and seems flexible enough for now.

**Comment by wing328** [comment]
@fehguy I agree and it's definitely better than the hard-coded `Petstore`. (my intention was to submit another PR soon with a better default package name)

On top of that, I suggest we come up with a better default package name base on the domain name from the `host`. (e.g. api.foursqaure.com maps to foursquare-php, foursquare-ruby, etc). The goal is to save time from users to come up with their own package naming convention (assuming the default one is good 90% of the time). In terms of priority, this is more like a day-2 requirement.

**Comment by gianluca-sabena** [comment]
+1

**Comment by hyeghiazaryan** [comment]
I am planning on implementing this. Has any work been already done/committed ?

**Comment by olensmar** [comment]
not that I know of - go for it :-)

**Comment by fehguy** [comment]
@hyeghiazaryan sounds great.  Basically we need to have a mechanism to pass args into the different language implementations and having them self-documented.  Then the -h command should be able to list the args for each of the implementations.

I believe we want to have both a command-line (arg) model as well as a JSON payload.  So for example setting the package for the models in the java generator could be something like such:

```
--model-package io.swagger.models
```

and in a configuration:
```json
{
   "modelPackage": "io.swagger.models"
}
```

**Comment by rastaman** [comment]
Hi, i'm wondering why there is no getters and setters on the members of the codegenerators classes, is there any reason ? 

I've patched my copy of swagger-codegen to have getters and setters for the JavaClient, DefaultCodeGen and JaxRSServer generators, and patched the swagger-maven-plugin to set them on the CodeGenConfig object and it works well and is easy to code. 

So i'm wondering what i have missed, could you give me a hint ? there is another way to inject properties without patching the code ? 

(Sorry but i just begin to play with the swagger codegen) 
PS : If anyone would like to take a look i have forked the repos (swagger-codegen and swagger-codegen-plugin) on my account, and the result works well for my use cases. I'm ok to push a pull request but would like to know before why it hasn't been already done :-)

**Comment by webron** [comment]
@rastaman - it would be much better to open an issue for your question that post it on a completely unrelated one.

**Comment by rastaman** [comment]
@webron : yup, sorry to have spammed this issue, in my mind it is closely related, i have open issue #779 . Have a good day

**Comment by hyeghiazaryan** [comment]
Work to add lang specific command-line args has been done.
Some args for java have been added. 
I would highly appreciate any feedback on the code/design.
Please see the last 7 commits here https://github.com/hyeghiazaryan/swagger-codegen/commits/develop_2.0

**Comment by hyeghiazaryan** [comment]
oops, seems like I have made changes to a Deprecated class, will transfer to swagger-codegen-cli

**Comment by hyeghiazaryan** [comment]
rolled back deprecated class changes
added --config option to swagger-cli along with handling
https://github.com/hyeghiazaryan/swagger-codegen/commits/develop_2.0

**Comment by wing328** [comment]
I think this can be closed as #805 has been merged.
